catalyst-gateway/bin/src/service/common/types/cardano/cip19_shelley_address.rs

```diff
@@ -0,0 +1,144 @@
+//! Cardano address types.
+//!
+//! More information can be found in [CIP-19](https://cips.cardano.org/cip/CIP-19)
+
+use std::{
+    borrow::Cow,
+    ops::{Deref, DerefMut},
+    sync::LazyLock,
+};
+
+use const_format::concatcp;
+use pallas::ledger::addresses::{Address, ShelleyAddress};
+use poem_openapi::{
+    registry::{MetaExternalDocument, MetaSchema, MetaSchemaRef},
+    types::{Example, ParseError, ParseFromJSON, ParseFromParameter, ParseResult, ToJSON, Type},
+};
+use serde_json::Value;
+
+use crate::service::common::types::string_types::impl_string_types;
+
+/// Title
+const TITLE: &str = "Cardano Payment Address";
+/// Description
+const DESCRIPTION: &str = "Cardano Shelley Payment Address (CIP-19 Formatted).";
+/// Example
+// cSpell:disable
+const EXAMPLE: &str = "addr_test1qz2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer3n0d3vllmyqwsx5wktcd8cc3sq835lu7drv2xwl2wywfgs68faae";
+// cSpell:enable
+/// Production Address Identifier
+const PROD: &str = "addr";
+/// Test Address Identifier
+const TEST: &str = "addr_test";
+/// Bech32 Match Pattern
+const BECH32: &str = "[a,c-h,j-n,p-z,0,2-9]";
+/// Length of the encoded address.
+const ENCODED_STAKED_ADDR_LEN: usize = 98;
+/// Length of the encoded address.
+const ENCODED_UNSTAKED_ADDR_LEN: usize = 53;
```

Comment:
Should we support pointer delegation part

Reply:
No, it's not even supported by Cardano anymore.